torch/_subclasses/functional_tensor.py

```diff
@@ -306,8 +316,23 @@ def unwrap(x):
             try:
                 # By default for python functionalization (for AOTAutograd), we reapply views.
                 old_apply_views = torch._functionalize_enable_reapply_views(True)  # type: ignore[attr-defined]
-                outs_unwrapped = func(*args_unwrapped, **kwargs_unwrapped)
-                outs_wrapped = pytree.tree_map_only(torch.Tensor, wrap, outs_unwrapped)
+
+                # Sometimes these functions cannot be directly dispatched to functionalize key
+                # because args are sometimes not functional tensors for some reason?
```

Comment:
🤔 can you make a repro - it's worth figuring out why this is happening

Reply:
It is actually following test case:
```
    def test_preserve_random(self):
        def fn(x):
            return torch.nn.functional.dropout(x, 0.5) + x

        x = torch.randn(4)

        torch.manual_seed(0)
        ref = fn(x)

        torch.manual_seed(0)
        aot_fn = aot_function(fn, nop)
        res = aot_fn(x)

        assert torch.allclose(ref, res)
  ```
  If I breakpoint in this branch, i see: 
  ```
  -> flat_f_args = pytree.tree_map(_to_fun, flat_args)
  /data/users/tmanlaibaatar/pytorch/torch/utils/_pytree.py(732)tree_map()
-> return treespec.unflatten(map(func, *flat_args))
  /data/users/tmanlaibaatar/pytorch/torch/utils/_pytree.py(599)unflatten()
-> leaves = list(leaves)
  /data/users/tmanlaibaatar/pytorch/torch/_functorch/_aot_autograd/collect_metadata_analysis.py(88)_to_fun()
-> r = to_fun(t)
  /data/users/tmanlaibaatar/pytorch/torch/_functorch/_aot_autograd/functional_utils.py(34)to_fun()
-> return FunctionalTensor.to_functional(t)
  /data/users/tmanlaibaatar/pytorch/torch/_subclasses/functional_tensor.py(168)to_functional()
-> x_functional = torch._to_functional_tensor(x)
> /data/users/tmanlaibaatar/pytorch/torch/_subclasses/functional_tensor.py(331)__torch_dispatch__()
-> outs_unwrapped = func(*args_unwrapped, **kwargs_unwrapped)
(Pdb) args_unwrapped
(FakeTensor(..., size=(4,)),)
(Pdb) func
<OpOverload(op='prim.device', overload='default')>
(Pdb) args_unwrapped
(FakeTensor(..., size=(4,)),)
(Pdb) 
```
Somehow torch._to_functional_tensor is invoking python FunctionalTensorMode again?

Reply:
I think we should do this!